File: typescript/src/Rover.ts
export interface Position {
    x: number,
    y: number
}

export enum OrientationEnum {
    N = 'N',
    E = 'E',
    S = 'S',
    W = 'W'
}

export interface WorldMap {
    height: number;
    width: number;
    rocks: Position[];
}

export interface RoverResult {
    message: string,
    position: Position,
}

export type Command = "f" | "b" | "l" | "r"

export class Rover {
    constructor(public position: Position, public direction: OrientationEnum, public worldMap: WorldMap) {
    }

    public move(commands: Command[]): RoverResult {
        return {message: 'commands done', position: this.position}
    }
}

File: go/rover.go
package _go

type Position struct {
	x int
	y int
}

const (
	N string = "N"
	E        = "E"
	S        = "S"
	W        = "W"
)

type WorldMap struct {
	height int
	width  int
	rocks  []Position
}
type RoverResult struct {
	message  string
	position Position
}

type Rover struct {
	position    Position
	orientation string
	worldMap    WorldMap
}

func (rover Rover) move(commands []string) RoverResult {
	return RoverResult{message: "commands done", position: rover.position}
}
